File: topicos/poo/java/ex1/Livro.java
import java.util.ArrayList;
import java.util.List;

public class Livro {
  private String ISBN;
  String titulo, autor;
  List<Leitor> leitores;

  public Livro(String ISBN, String titulo, String autor) {
    this.ISBN = ISBN;
    this.titulo = titulo;
    this.autor = autor;
    this.leitores = new ArrayList<>();
  }

  public void putLeitor(Leitor leitor) {
    this.leitores.add(leitor);
  }

  public void printLeitores() {
    for (Leitor leitor : this.leitores) {
      System.out.println(leitor.nome + " esta lendo " + this.titulo);
    }
  }
}

File: topicos/poo/go/ex1/livro.go
package biblioteca

import "fmt"

type Livro struct {
	isbn     string
	Titulo   string
	Autor    string
	Leitores []Leitor
}

func NewLivro(isbn, titulo, autor string) *Livro {
	return &Livro{
		Titulo:   titulo,
		Autor:    autor,
		isbn:     isbn,
		Leitores: make([]Leitor, 0),
	}
}

func (l Livro) incluirLeitor(leitor Leitor) {
	l.Leitores = append(l.Leitores, leitor)
}

func (l Livro) imprimirLeitores() {
	for i := 0; i < len(l.Leitores); i++ {
		fmt.Printf("\n%s leu %d", l.Leitores[i].Nome, l.Titulo)
	}
}
